sky/services/raw_keyboard/src/org/chromium/mojo/raw_keyboard/RawKeyboardServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.chromium.mojo.raw_keyboard;
+
+import org.chromium.mojo.system.MojoException;
+import org.chromium.mojom.raw_keyboard.RawKeyboardListener;
+import org.chromium.mojom.raw_keyboard.RawKeyboardService;
+
+/**
+ * Android implementation of Keyboard.
+ */
+public class RawKeyboardServiceImpl implements RawKeyboardService {
+    // We have a unique ServiceImpl per connection.  However multiple views
+    // can be sending us key events simultaneously.
+    private static RawKeyboardServiceState sViewState;
```

Comment:
This is going to give you some grief.  You need to know what views those key events came from so that you can demultiplex them.